Project-Hotpot/Project-Hotpot/Managers/SpotifyManager.swift

```diff
@@ -199,14 +199,56 @@ extension SpotifyManager {
         
     }
     
-    func fetchAudioFeaturesFromTracks (for trackIDs: [String], completion: @escaping (_ result: Result<[String: Any], Error>) -> Void) {
+    func fetchAudioFeaturesFromTracks (for trackIDs: [String], completion: @escaping (_ result: Result<[String:Any], Error>) -> Void) {
         let trackIdsString = trackIDs.joined(separator: ",")
         guard let url = URL(string: "https://api.spotify.com/v1/audio-features?ids=" + trackIdsString) else {
             return completion(.failure(CustomError.invalidURL))
         }
         createRequest(url: url, completion: completion)
     }
     
+     func spotifyIdToSongDetails(ids: [String], completion: @escaping (_ result: Result<[SongDetails], Error>) -> Void) {
```

Comment:
Nice! 
Just giving you more info about newer design patterns for the future. Sometimes logic like this means that now manager needs to know about SongDetails. So here we have somewhat a cyclic dep. Specifically in this case, this is very much suited for a "Converter Pattern"  (See: https://java-design-patterns.com/patterns/converter/)